src/Sarif/ExtensionMethods.cs

```diff
@@ -441,6 +441,39 @@ internal static string PropertyValue(this Dictionary<string, string> properties,
             return null;
         }
 
+        /// <summary>
+        /// Merge this property bag with another one, preferring the properties in this property
+        /// bag if there are any duplicates.
+        /// </summary>
+        /// <param name="propertyBag">
+        /// The property bag into which <paramref name="otherPropertyBag"/> is to be merged.
+        /// </param>
+        /// <param name="otherPropertyBag">
+        /// A property bag containing properties to merge into <paramref name="propertyBag"/>.
+        /// </param>
+        /// <typeparam name="T">
+        /// The type of object in the property bag. For SARIF property bags, the type will
+        /// be <see cref="SerializedPropertyInfo"/>.
+        /// </typeparam>
+        /// <returns>
+        /// The original <paramref name="propertyBag"/> object, into which the properties
+        /// from <paramref name="otherPropertyBag"/> have now been merged.
+        /// </returns>
+        public static IDictionary<string, T> MergePreferFirst<T>(
```

Comment:
>T [](start = 62, length = 1)

Type parameter needed because the `GitHubDspIngestionVisitor` uses this method with SARIF property bags -- so `T` is `SerializedPropertyInfo` -- whereas the result matching code where I found this was using it with a dictionary whose values were of type `object`.